File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotutil/Direction.java
package org.firstinspires.ftc.teamcode.robotutil;

/**
 * Created by
 * Anton on 1/10/17.
 */

public enum Direction {
    FORWARD(1),BACK(-1),CW(1),CCW(-1),RIGHT(1),LEFT(-1),UP(1),DOWN(-1);

    private final int dir;
    Direction(int dir) { this.dir = dir; }
    public int getValue() { return dir; }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Models/Direction.kt
package org.firstinspires.ftc.teamcode.Models

enum class Direction(val intRepr: Int) {
    FORWARD(1),
    BACKWARD(-1),
    SPIN_CW(1),
    SPIN_CCW(-1),
    FORWARD_LEFT(1),
    FORWARD_RIGHT(-1),
    BACKWARD_LEFT(1),
    BACKWARD_RIGHT(-1)
}